### ISSUE TITLE
The Doc is misleading

### ISSUE THREAD

**Issue by ueen** [leandroBorgesFerreira/LoadingButtonAndroid#133]
First, it says to use an ImageButton to created in the XML and then casts a Button, this causes the app to crash if you follow the sample code!!

**Comment by leandroBorgesFerreira** [comment]
That's true =/. I'll fix that this week.

**Comment by ueen** [comment]
Also, there seems to be an issue with startAnimation(*) it requires a storage *Function<> i can not provide (implemented according to docs with Java in AndroidStudio with AndroidX, everything uptodate) :/

**Comment by leandroBorgesFerreira** [comment]
Fixed the documentation. Can you provide the code that you are trying to use?

**Comment by ueen** [comment]
Ok, closely followed the doc on a clean Project in latest Android Studio (Java), AnroidX, everything uptodate, 

startAnimation() gets a "Cannot resolve" Annotation

This is the compile errormessage:

error: method revertAnimation in class CircularProgressButton cannot be applied to given types;
required: Function0<Unit>
found: no arguments
reason: actual and formal argument lists differ in length

put Button according to doc in XML (works on its own)
then in onCreateView (Fragment) i put this code

final CircularProgressButton btn = (CircularProgressButton) v.findViewById(R.id.btn_id);
        new CountDownTimer(5000, 1000) {
            @Override
            public void onTick(long millisUntilFinished) {
                btn.startAnimation();
            }

            @Override
            public void onFinish() {
                btn.revertAnimation();
            }
        }.start();

**Comment by leandroBorgesFerreira** [comment]
This will work: 
```
final CircularProgressButton btn = findViewById(R.id.buttonTest1);
        new CountDownTimer(5000, 1000) {
            @Override
            public void onTick(long millisUntilFinished) {
                btn.startAnimation(() -> null);
            }

            @Override
            public void onFinish() {
                btn.revertAnimation(() -> null);
            }
        }.start();
```

**Comment by leandroBorgesFerreira** [comment]
This library was optimized to Kotlin, sorry do you have to use those callbacks... Java doesn't recognize the default values

**Comment by ueen** [comment]
`btn.startAnimation(() -> null);`
Lambda expressions are not supported at language level '7'
Also
`btn.startAnimation(null);`
Ambiguous method call. Both
startAnimation
(Function0<Unit>)
in CircularProgressButton and
startAnimation
(Animation)
in View match

**Comment by ueen** [comment]
One more issue tough, after reverting the Animation, the Button loses its original rounded corners, is there a way to get the Button back its original idle state?

**Comment by leandroBorgesFerreira** [comment]
Yep, there is. Try: `app:finalCornerAngle`. That's the one if I remember correctly.

**Comment by leandroBorgesFerreira** [comment]
Or maybe initialCornerAngle... One of the two.

**Comment by ardevd** [comment]
When I try to do `button.startAnimation()`

```
MyFragment.java:197: error: cannot access Function0
                mProgressButton.startAnimation(null);
                               ^
  class file for kotlin.jvm.functions.Function0 not found
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

**Comment by ardevd** [comment]
Using lambdas also result in errors for me: 
```
ragment.java:197: error: cannot access Function0
                mProgressButton.startAnimation(() -> null);
                               ^
  class file for kotlin.jvm.functions.Function0 not found
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

**Comment by ueen** [comment]
Both also changes the loading shape, which is not what I want, as it should stay round.

**Comment by minhduc0711** [comment]
Hi, I am also using Java and I ran into another problem at this line: 
`btn.startAnimation(() -> null);`
`Cannot resolve method 'startAnimation(<lambda expression>)`

Can you help me? Thanks.
